src/02-common-principles.md

```diff
@@ -383,6 +383,286 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## BIDS URI: Pointing to files within and outside of BIDS datasets
+
+Throughout the BIDS specification there are metadata fields that contain one or more
+pointers to files
+(for example `IntendedFor`, `SpatialReference`, and `AssociatedEmptyRoom`).
+Prior to BIDS version 1.7.0, such pointers were often required to be "relative paths".
+However the specification failed to consistently describe *what* these pointers should be relative to,
+and the exact syntax to be used.
+Yet even with consistently defined "relative paths",
+this method of pointing to files becomes impractical when referring to files that
+are outside of a particular file tree (for example, on the Internet),
+or when the location of the data that is being pointed to changes.
+
+For these reasons, starting with BIDS version 1.7.0,
+using "relative paths" for all fields pointing to a separate file is [DEPRECATED][].
+For internal consistency, the use of "absolute paths" is [DEPRECATED][] as well.
+
+Instead, the following [URI][uniform-resource-indicator]
+scheme SHOULD be used to point to files within and outside of BIDS datasets,
+as elaborated below.
+
+`bids:<dataset-name>:/absolute/path/within/dataset`
+
+Here, `bids:` is the URI scheme,
+and `<dataset-name>:/absolute/path/within/dataset` is the path to the file,
+consisting of the name of a dataset (MAY be an empty string, as defined below), a colon,
+and the absolute path within that dataset that MUST start with a forward slash `/`.
+
+The location of a dataset with the name `<dataset-name>` MUST be specified in the
+[`dataset_description.json` file](./03-modality-agnostic-files.md#dataset-description)
+under the `DatasetLinks` field,
+which is an [object][] of [URIs][], as shown in the examples below.
+
+In brief,
+`dataset_description.DatasetLinks` contains mappings from strings to URIs that point to dataset locations.
+The `<dataset-name>`: `""` (that is, an empty string)
+is a reserved value that may only be used to refer to the *root of the current dataset*;
+an empty string (`""`) MUST NOT be a key in the `DatasetLinks` object.
+
+In the context of BIDS URIs,
+a dataset with a given `<dataset-name>` does not have to be a BIDS dataset,
+or a neuroimaging dataset.
+It may just as well be a (sub-)directory containing one or more arbitrary files.
+
+In the case where a derivatives dataset is nested under a raw dataset and both have a `dataset_description.json` file,
+the BIDS URIs within the nested derivatives dataset MUST be resolved with respect to `/derivatives/dataset_description.json`,
```

Comment:
Generally we don't treat `<bids>/derivatives/` itself as a derivative dataset but a container for derivative datasets.

```suggestion
the BIDS URIs within the nested derivatives dataset MUST be resolved with respect to `/derivatives/<derivative>/dataset_description.json`,
```